addons/web/static/src/js/views/form_view.js

```diff
@@ -765,6 +770,33 @@ var FormView = View.extend(common.FieldManagerMixin, {
             });
         });
     },
+    on_button_archive_unarchive: function(event) {
+        var self = this;
+        var active_field_value;
+        var def = $.Deferred();
+        var write_value = function(active_field_value) {
+            self.dataset.write(self.datarecord.id, {'active': active_field_value}).done(function() {
+                if (self.dataset.size()) {
+                    self.execute_pager_action('next');
+                } else {
+                    self.do_action('history_back');
+                }
+                def.resolve();
+            }).fail(function() {
+                def.reject();
+            });
+        };
+        this.has_been_loaded.done(function() {
+            if (self.get_field_value('active') === true) {
```

Comment:
is it really required `===`  just `if(self.get_field_value('active'))`  is not enough because active is boolean filed ?